File: backend/src/main/java/com/control/inventory/entities/Movimentacao.java
package com.control.inventory.entities;

import com.control.inventory.entities.enums.TipoMovimento;
import lombok.*;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Date;

@Entity
@Table(name="tb_estoque_movimento")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Movimentacao implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @JoinColumn(name = "tipo_movimento")
    private TipoMovimento tipoMovimento;
    private Integer quantidade;
    @DateTimeFormat(pattern = "dd/MM/yyyy")
    private Date data;
    private String motivo;
    private String documento;

    private BigDecimal saldo;

    private String situacao;

    @ManyToOne
    @JoinColumn(name = "produto_id")
    private Produto produto;

}

File: frontend/src/app/common/movimentacao.ts
import { Produto } from './produto';
import { TipoMovimento } from './tipoMovimento';

export interface Movimentacao {
  id: number;
  produtoId: number;
  tipoMovimento: TipoMovimento;
  data: Date;
  documento: string;
  motivo: string;
  quantidade: number;
  saldo: number;
  situacao: string;
}
